nilaway_test.go

```diff
@@ -25,205 +25,57 @@ import (
-	analysistest.Run(t, testdata, Analyzer, "go.uber.org/consts")
-}
+func TestNilAway(t *testing.T) {
+	t.Parallel()
+
+	testdata := analysistest.TestData()
+
+	// For descriptions of the purpose of each of the following tests, consult their source files
+	// located in testdata/src/<package>.
+
+	tests := []struct {
+		name     string
+		patterns []string
+	}{
+		{name: "Inference", patterns: []string{"go.uber.org/inference"}},
+		{name: "Contracts", patterns: []string{"go.uber.org/contracts", "go.uber.org/contracts/namedtypes"}},
+		{name: "Testing", patterns: []string{"go.uber.org/testing"}},
+		{name: "ErrorReturn", patterns: []string{"go.uber.org/errorreturn", "go.uber.org/errorreturn/inference"}},
+		{name: "Maps", patterns: []string{"go.uber.org/maps"}},
+		{name: "Slices", patterns: []string{"go.uber.org/slices", "go.uber.org/slices/inference"}},
+		{name: "Arrays", patterns: []string{"go.uber.org/arrays"}},
+		{name: "Channels", patterns: []string{"go.uber.org/channels"}},
+		{name: "GoQuirks", patterns: []string{"go.uber.org/goquirks"}},
+		{name: "GlobalVars", patterns: []string{"go.uber.org/globalvars"}},
+		{name: "DeepNil", patterns: []string{"go.uber.org/deepnil", "go.uber.org/deepnil/inference"}},
+		{name: "NilableTypes", patterns: []string{"go.uber.org/nilabletypes"}},
+		{name: "HelloWorld", patterns: []string{"go.uber.org/helloworld"}},
+		{name: "MultiFilePackage", patterns: []string{"go.uber.org/multifilepackage", "go.uber.org/multifilepackage/firstpackage", "go.uber.org/multifilepackage/secondpackage"}},
+		{name: "MultipleAssignment", patterns: []string{"go.uber.org/multipleassignment"}},
+		{name: "AnnotationParse", patterns: []string{"go.uber.org/annotationparse"}},
+		{name: "NilCheck", patterns: []string{"go.uber.org/nilcheck"}},
+		{name: "SimpleFlow", patterns: []string{"go.uber.org/simpleflow"}},
+		{name: "LoopFlow", patterns: []string{"go.uber.org/loopflow"}},
+		{name: "MethodImplementation", patterns: []string{"go.uber.org/methodimplementation", "go.uber.org/methodimplementation/mergedDependencies", "go.uber.org/methodimplementation/chainedDependencies", "go.uber.org/methodimplementation/multipackage", "go.uber.org/methodimplementation/embedding"}},
+		{name: "NamedReturn", patterns: []string{"go.uber.org/namedreturn"}},
+		{name: "IgnoreGenerated", patterns: []string{"go.uber.org/ignoregenerated"}},
+		{name: "IgnorePackage", patterns: []string{"ignoredpkg1", "ignoredpkg2"}},
+		{name: "Receivers", patterns: []string{"go.uber.org/receivers", "go.uber.org/receivers/inference"}},
+		{name: "Generics", patterns: []string{"go.uber.org/generics"}},
+		{name: "FunctionContracts", patterns: []string{"go.uber.org/functioncontracts", "go.uber.org/functioncontracts/inference"}},
+		{name: "Constants", patterns: []string{"go.uber.org/consts"}},
+		{name: "ErrorMessage", patterns: []string{"go.uber.org/errormessage"}},
+	}
 
-func TestErrorMessage(t *testing.T) {
-	t.Parallel()
+	for _, tt := range tests {
+		tt := tt
```

Comment:
Is this shadowing really needed?

Reply:
Yes, it's a common pattern for Go, since loop variable is per loop not per iteration, so we must shadow the variable such that the parallel test is accessing the correct `tt`.

Luckily, Go finally got rid of this weirdness in Go 1.22: https://tip.golang.org/doc/go1.22 so we don't have to use this trick once we require Go 1.22 (i.e., in another 6 months) :)